plugins/modules/server_info.py

```diff
@@ -0,0 +1,240 @@
+from ansible.module_utils._text import to_native
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community',
+}
+USER_AGENT = 'ansible-module/%s_ionos-cloud-sdk-python/%s' % (__version__, sdk_version)
+DOC_DIRECTORY = 'compute-engine'
+STATES = ['info']
+OBJECT_NAME = 'Servers'
+
+OPTIONS = {
+    'datacenter': {
+        'description': ['The ID of the datacenter.'],
+        'available': STATES,
+        'required': STATES,
+        'type': 'str',
+    },
+    'upgrade_needed': {
+        'description': ['Filter servers that can or that cannot be upgraded.'],
+        'available': STATES,
+        'type': 'bool',
+    },
+    'api_url': {
+        'description': ['The Ionos API base URL.'],
+        'version_added': '2.4',
+        'env_fallback': 'IONOS_API_URL',
+        'available': STATES,
+        'type': 'str',
+    },
+    'username': {
+        # Required if no token, checked manually
+        'description': ['The Ionos username. Overrides the IONOS_USERNAME environment variable.'],
+        'aliases': ['subscription_user'],
+        'env_fallback': 'IONOS_USERNAME',
+        'available': STATES,
+        'type': 'str',
+    },
+    'password': {
+        # Required if no token, checked manually
+        'description': ['The Ionos password. Overrides the IONOS_PASSWORD environment variable.'],
+        'aliases': ['subscription_password'],
+        'available': STATES,
+        'no_log': True,
+        'env_fallback': 'IONOS_PASSWORD',
+        'type': 'str',
+    },
+    'token': {
+        # If provided, then username and password no longer required
+        'description': ['The Ionos token. Overrides the IONOS_TOKEN environment variable.'],
+        'available': STATES,
+        'no_log': True,
+        'env_fallback': 'IONOS_TOKEN',
+        'type': 'str',
+    },
+}
+
+
+def transform_for_documentation(val):
+    val['required'] = len(val.get('required', [])) == len(STATES)
+    del val['available']
+    del val['type']
+    return val
+
+
+DOCUMENTATION = '''
+---
+module: server_info
+short_description: List Ionos Cloud servers of a given datacenter.
+description:
+     - This is a simple module that supports listing servers.
+version_added: "2.0"
+options:
+''' + '  ' + yaml.dump(
+    yaml.safe_load(str({k: transform_for_documentation(v) for k, v in copy.deepcopy(OPTIONS).items()})),
+    default_flow_style=False).replace('\n', '\n  ') + '''
+requirements:
+    - "python >= 2.6"
+    - "ionoscloud >= 6.0.2"
+author:
+    - "IONOS Cloud SDK Team <sdk-tooling@ionos.com>"
+'''
+
+EXAMPLES = '''
+    - name: Get all servers for given datacenter
+      server_info:
+        datacenter: "{{ datacenter }}"
+        register: server_list_response
+
+    - name: Get only the servers that need to be upgraded
+      server_info:
+        datacenter: "{{ datacenter }}"
+        register: servers_list_upgrade_response
+        upgradeNeeded: true
```

Comment:
I think this should be snake case

Reply:
good catch, now i also saw that register should be 1 tab to the left, thank you

Reply:
you're welcome